astropy/cosmology/_utils.py

```diff
@@ -81,3 +85,29 @@ def all_cls_vars(obj: object | type, /) -> dict[str, Any]:
     """Return all variables in the whole class hierarchy."""
     cls = obj if isinstance(obj, type) else obj.__class__
     return functools.reduce(operator.__or__, map(vars, cls.mro()[::-1]))
+
+
+def all_parameters(obj: object | type, /) -> dict[str, Field | Parameter]:
+    """Get all fields of a dataclass, including those not-yet finalized.
+
+    Parameters
+    ----------
+    obj : object | type
+        A dataclass.
+
+    Returns
+    -------
+    dict[str, Field | Parameter]
+        All fields of the dataclass, including those not yet finalized in the class, if
+        it's still under construction, e.g. in ``__init_subclass__``.
+    """
+    from astropy.cosmology.parameter import Parameter
```

Comment:
Is there an import loop?

Reply:
Yeah, there was. Utils should not depend on anything else.